VirtualFilesystemTestTrait.php

```diff
@@ -63,9 +63,21 @@ trait VirtualFilesystemTestTrait {
 	public function init() {
 		if ( empty( $this->config ) ) {
 			$this->loadConfig();
+
+			if ( ! isset( $this->config['vfs_dir'] ) || false === $this->config['vfs_dir'] ) {
+				$this->config['vfs_dir'] = '';
+			} elseif ( '' !== $this->config['vfs_dir'] ) {
+				$this->config['vfs_dir'] = rtrim( $this->config['vfs_dir'], '\//' ) . '/';
+			}
+		}
+
+		if ( '' === $this->config['vfs_dir'] ) {
+			$vfs = null;
+		} else {
+			$vfs = trim( $this->config['vfs_dir'], '/' );
```

Comment:
This will break a lot of the tests that are already written. Why do we need to remove the `/`?

Reply:
The current code already does this:
```php
$vfs = ArrayTrait::get( $this->config['structure'], rtrim( $this->config['vfs_dir'], '\//' ), [], '/' );
```
I only changed the 2nd argument of `rtrim()` for consistency since trimming backslashes is not needed.

Reply:
Except that the tests themselves use the given value in `'vfs_dir'`. It's used for more than just this part of the code.

Reply:
Except that I don't change the value of `$this->config['vfs_dir']` here.

Reply:
Right. It's on lines 68 and 70 where you're overwriting it.

Let me ask, what is the purpose of this code? Why do we need it? 

Instead, to get to the root, you set the configuration in the test data file to:

```php

return [
    'vfs_dir' => '',
...
];
```

Reply:
Because of `ArrayTrait::get()`, notice the `is_null( $keyToFind )`:
```php
	public static function get( $search, $keyToFind, $default = null, $delimiter = '/' ) {
		if ( ! is_array( $search ) ) {
			return $default;
		}

		if ( is_null( $keyToFind ) ) {
			return $search;
		}

		if ( array_key_exists( $keyToFind, $search ) ) {
			return $search[ $keyToFind ];
		}

		foreach ( explode( $delimiter, $keyToFind ) as $segment ) {
			if ( is_array( $search ) && array_key_exists( $segment, $search ) ) {
				$search = $search[ $segment ];
			} else {
				return $default;
			}
		}

		return $search;
	}
```

Reply:
Okay, let's walk through this `get()` method using the scenario of:

```php
return [
    'vfs_dir' => '',
...
];
```

and:

```php
ArrayTrait::get( $this->config, 'vfs_dir' );
```

What will happen? I'll add comments to the code to help us figure it out.

```php
public static function get( $search, $keyToFind, $default = null, $delimiter = '/' ) {
	// Yes, $search is an array. 
	if ( ! is_array( $search ) ) {
		return $default;
	}

	// The key = 'vfs_dir'.
	if ( is_null( $keyToFind ) ) {
		return $search;
	}

	// The key does exist in the search array. So it returns the empty string.
	if ( array_key_exists( $keyToFind, $search ) ) {
		return $search[ $keyToFind ];
	}

	foreach ( explode( $delimiter, $keyToFind ) as $segment ) {
		if ( is_array( $search ) && array_key_exists( $segment, $search ) ) {
			$search = $search[ $segment ];
		} else {
			return $default;
		}
	}

	return $search;
}
```

What do you think @Screenfeed?

Reply:
@hellofromtonya Yes, this is also the behavior I would expect, but I don't see this case used anywhere.
The only case I found using `ArrayTrait::get()` is this one:
```php
$vfs = ArrayTrait::get( $this->config['structure'], rtrim( $this->config['vfs_dir'], '\//' ), [], '/' );
```
Which translates like:
```php
$vfs = ArrayTrait::get( [ ...structure... ], '', [], '/' );
```
So it will return an empty array instead of the whole structure.

Reply:
You're right @Screenfeed. Very good point. 

I expanded upon this problem and pushed some new commits to fix it plus:

- define the default config structure
- allow a test case to skip building the original files and dirs properties (not always needed)